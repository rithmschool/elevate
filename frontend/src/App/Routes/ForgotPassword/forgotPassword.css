.forgotPassword_container {
  background-color: #f4f6f8;
  margin-top: 10%;
}

.form-div {
  text-align: "center";
}

.form-title {
  margin-top: 2%;
}