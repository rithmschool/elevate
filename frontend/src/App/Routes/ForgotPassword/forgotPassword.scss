.forgotPassword-container {
  background-color: #f4f6f8;
  margin-top: 10%;
  padding-bottom: 5px;

  .text {
    text-align: center;

    .title {
      margin-top: 5%;
    }
  }

  .button {
    display: flex;
    justify-content: center;
    margin-bottom: 5%;
  }
}