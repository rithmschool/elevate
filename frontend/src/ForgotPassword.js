import React from 'react';
import axios from 'axios'
import { Col, Button, Form, Label, Input, Row} from 'reactstrap';

class ForgotPassword extends React.Component {

  constructor(props){
    super(props);
    this.state = {
      email: '',
      messageFromServer: '',
      showError: false,
    }
    this.handleChange = this.handleChange.bind(this);
    this.sendEmail = this.sendEmail.bind(this);
  }


  handleChange(evt){
		this.setState({ [evt.target.name]: evt.target.value });
  }

  sendEmail(evt){
    evt.preventDefault();
    if(this.state.email === ''){
      this.setState({
        showError: false, messageFromServer: '',
      });
    } else {
      axios.post('http://localhost:3001/forgotpassword', {
        email: this.state.email,
      })
      .then(response => {
        console.log(response.data);
        if(response.data === 'email not in db'){
          this.setState({
            showError: false, messageFromServer: ''
          });
        } else if(response.data === 'recovery email sent'){
            this.setState({
              showError: false,
              messageFromServer: 'recovery email senf',
            });
        }
      })
      .catch(error => {
        console.log(error.data);
      });
    }
  }

  render(){
    const { email, messageFromServer, showError} = this.state;
    return(
      <div>
        <form onSubmit={this.sendEmail}>
        <Row form>
          <Col md={4}>                  
            <Label className="form-group has-float-label">
              <Input onChange={this.handleChange}
                    value={email}
                    type="email"
                    name="email"
                    id="EditUser-first_name"
                    />
              <span>Email</span>
            </Label>
            </Col>
          </Row>
          <Col >
            <Button color="info" size="sm"> 
              Send Password Reset Email
            </Button>
        </Col>
        </form>
      </div>
    )
  }
}
export default ForgotPassword;