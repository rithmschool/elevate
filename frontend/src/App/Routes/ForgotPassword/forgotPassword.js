import React from "react";
import { Col, Button, Form, Label, Input, Row } from "reactstrap";
import ElevateApi from "../../../elevateApi";
import Alert from "../Alert/alert";
import Spinner from "../../Spinner/spinner";
import "./forgotPassword.css";

class ForgotPassword extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: "",
      errors: [],
      msgFromServer: [],
      emailSent: false,
      isLoading: false
    };

    this.handleChange = this.handleChange.bind(this);
    this.sendEmail = this.sendEmail.bind(this);
  }

  handleChange(evt) {
    this.setState({
      [evt.target.name]: evt.target.value,
      emailSent: false,
      errors: []
    });
  }

  async sendEmail(evt) {
    evt.preventDefault();

    if (this.state.email === "") {
      this.setState({ errors: ["Please enter your email address!"] });
    } else {
      try {
        this.setState({ isLoading: true });

        let response = await ElevateApi.forgotPassword({
          email: this.state.email
        });

        this.setState({
          errors: [],
          emailSent: true,
          msgFromServer: [response.message],
          isLoading: false
        });
      } catch (errors) {
        this.setState({ errors, isLoading: false });
      }
    }
  }

  render() {
    const { email, errors, emailSent, isLoading } = this.state;

    if (isLoading) return <Spinner />;

    return (
      <div
        className={`
          container
          col-md-6
          offset-md-3
          col-lg-4
          offset-lg-4
          border
          rounded
          shadow
          forgotPassword-container`}
      >
        <Form onSubmit={this.sendEmail}>
          <div className="text-center">
            <h3 className="m-2">Forgot Password</h3>
            <p>
              Please enter your email address and we'll send you instructions on
              how to reset your password
            </p>
          </div>

          <Row form>
            <Col md={12}>
              <Label className="form-group has-float-label">
                <Input
                  onChange={this.handleChange}
                  value={email}
                  type="email"
                  name="email"
                  id="forgt-email"
                />
                <span>Email</span>
              </Label>
            </Col>
          </Row>

          {errors.length > 0 && <Alert type="danger" messages={errors} />}

          {emailSent && (
            <Alert type="success" messages={this.state.msgFromServer} />
          )}

          <Col align="center">
            <Button size="sm" className="btn btn-info">
              Send Password Recovery Email
            </Button>
          </Col>
        </Form>
      </div>
    );
  }
}

export default ForgotPassword;
