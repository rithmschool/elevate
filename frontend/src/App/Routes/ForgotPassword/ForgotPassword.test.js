import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import ForgotPassword from "./forgotPassword";

it("renders without crashing", function() {
  shallow(<ForgotPassword />);
});

it("matches snapshot", function() {
  let wrapper = shallow(<ForgotPassword />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

describe("Forgot password form", function() {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<ForgotPassword />);
  });

  it("has states", function() {
    expect(wrapper.state("email")).toEqual("");
    expect(wrapper.state("errors")).toEqual([]);
    expect(wrapper.state("emailSent")).toEqual(false);
  });

  it("test user entred an email", function() {
    wrapper.find("input").getDOMNode().value = "test@test.com";
    wrapper.find("input").simulate("change");
    wrapper.find("form").simulate("submit");

    expect(wrapper.state("email")).toEqual("test@test.com");
  });

  it('test user didn"t enter an email and try to submit', function() {
    wrapper.find("form").simulate("submit");
    expect(wrapper.state("errors")).toEqual([
      "Please enter your email address!"
    ]);
  });
});
