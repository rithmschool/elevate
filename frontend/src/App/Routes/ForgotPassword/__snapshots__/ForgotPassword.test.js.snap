// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="
          container
          col-md-6
          offset-md-3
          col-lg-4
          offset-lg-4
          border
          rounded
          shadow
          forgotPassword_container"
>
  <div>
    <Form
      inline={false}
      onSubmit={[Function]}
    >
      <div
        className="text"
      >
        <h3
          className="title"
        >
          Reset Password
        </h3>
        <p>
          Please enter your email address and we'll send you instructions on how to reset your password
        </p>
      </div>
      <FormGroup>
        <FormControl
          className="form-group has-float-label"
          id="forgt-email"
          name="email"
          onChange={[Function]}
          placeholder="email"
          type="email"
          value=""
        />
      </FormGroup>
      <div
        className="button"
      >
        <Button
          active={false}
          className="btn btn-info"
          disabled={false}
          size="sm"
          type="button"
          variant="primary"
        >
          Send Password Recovery Email
        </Button>
      </div>
    </Form>
  </div>
</div>
`;
